app/controllers/admin/instances_controller.rb

```diff
@@ -54,6 +54,12 @@ def set_instance
     end
 
     def set_instances
+      # If we have `limited` in the query parameters, remove it and redirect to suspended:
+      return redirect_to admin_instances_path filter_params.merge(status: :suspended) if params[:limited].present?
+
+      # If we're in limited federation mode and have a status parameter, remove it:
+      return redirect_to admin_instances_path filter_params.merge(status: nil) if limited_federation_mode? && params[:status].present?
+
```

Comment:
These feel like they're in the wrong place, but I want to do these redirects before assigning `@instances`

Reply:
These indeed feel like the wrong place. This probably should be moved to a separate `before_action` filter.